File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    answer = 0
    p_len = len(p)
    p_val = int(p)
    
    for i in range(len(t) - p_len + 1):
        sub_val = int(t[i : i + p_len])
        
        if sub_val <= p_val:
            answer += 1
            
    return answer

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.js
function solution(t, p) {
    let count = 0;
    const pLen = p.length;
    const pNum = Number(p);

    for (let i = 0; i <= t.length - pLen; i++) {
        const sub = t.slice(i, i + pLen);
        
        if (Number(sub) <= pNum) {
            count++;
        }
    }
    return count;
}